b2sdk/sync/sync.py

```diff
@@ -189,37 +195,24 @@ def sync_folders(self, source_folder, dest_folder, now_millis, reporter):
         queue_limit = self.max_workers + 1000
         sync_executor = BoundedQueueExecutor(unbounded_executor, queue_limit=queue_limit)
 
-        # First, start the thread that counts the local files. That's the operation
-        # that should be fastest, and it provides scale for the progress reporting.
-        local_folder = None
-        if source_folder.folder_type() == 'local':
-            local_folder = source_folder
-        if dest_folder.folder_type() == 'local':
-            local_folder = dest_folder
-        if local_folder is None:
-            raise ValueError('neither folder is a local folder')
-        if reporter:
-            sync_executor.submit(count_files, local_folder, reporter)
+        if source_type == 'local' and reporter is not None:
+            # Start the thread that counts the local files. That's the operation
+            # that should be fastest, and it provides scale for the progress reporting.
+            sync_executor.submit(count_files, source_folder, reporter, self.policies_manager)
 
         # Schedule each of the actions
         bucket = None
-        if source_folder.folder_type() == 'b2':
-            bucket = source_folder.bucket
-        if dest_folder.folder_type() == 'b2':
+        if dest_type == 'b2':
             bucket = dest_folder.bucket
-        if bucket is None:
-            raise ValueError('neither folder is a b2 folder')
-        total_files = 0
-        total_bytes = 0
+        elif source_type == 'b2':
+            bucket = source_folder.bucket
```

Comment:
we now have b2-to-b2 sync, so `bucket` is a very confusing name for a variable here. Please adjust it.

Reply:
I don't have a better candidate. Could you share if you have? It's a bucket on which we're performing API operations.

Reply:
`action_bucket`?

Reply:
Will change